subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -610,10 +687,23 @@ void ztest_verify_all_test_suites_ran(void)
 	}
 }
 
+void ztest_run(const void *state)
+{
+	if(list_tests) {
+		ztest_list_tests(state);
+	} else {
+		if(test_args == NULL) {
+			ztest_run_test_suites(state);
+			ztest_verify_all_test_suites_ran();
+		} else {
+			ztest_run_filtered_tests(state, test_args);
```

Comment:
Some tests need to have a custom `test_main` (our own chromium tests included). This makes that logic much more difficult to recreate. Can we not add this logic to `ztest_run_test_suites()`. I would think that it should be possible to:

1. Iterate on the suites and if there's a filter verify that the suite is in the filter before calling `z_ztest_run_test_suite_ptr()`
2. When running the tests in a suite, also check the filter per test.
3. For tests that are filtered, don't increment `failed` (like you have) but do increment `run_count`.

This should allow `ztest_verify_all_test_suites_ran()` to pass